### PR TITLE
feat(memory-v2): add router prompt template

### PR DESCRIPTION
**Pull request by siddseethepalli**
## Summary
- Add `renderRouterPrompt` in `assistant/src/memory/v2/prompts/router.ts` following the sweep prompt pattern.
- Substitutes `{{ASSISTANT_NAME}}`, `{{USER_NAME}}`, `{{PAGE_INDEX}}`; falls back to neutral names when null.
- Add tests for substitution, null fallback, empty index, and snapshot stability.

Part of plan: memory-v2-sonnet-router.md (PR 7 of 10)

---

<a href="https://app.devin.ai/review/vellum-ai/vellum-assistant/pull/30166" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open in Devin Review
  </picture>
</a>

### CODE REVIEW

**Comment by siddseethepalli** [comment]
Addressed in https://github.com/vellum-ai/vellum-assistant/pull/30539

File: assistant/src/memory/v2/prompts/router.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+ * still produces well-formed English when either name is unavailable on this
+ * workspace. The page index is substituted verbatim — callers are responsible
+ * for trimming/formatting it.
+ */
+export function renderRouterPrompt(opts: RenderRouterPromptOpts): string {
+  const assistant = opts.assistantName?.trim() || "the assistant";
+  const user = opts.userName?.trim() || "the user";
+  return ROUTER_PROMPT.replaceAll(ASSISTANT_NAME_PLACEHOLDER, assistant)
+    .replaceAll(USER_NAME_PLACEHOLDER, user)
+    .replaceAll(PAGE_INDEX_PLACEHOLDER, opts.pageIndexBlock);
```
[the remark this answers is not in the document]
Comment (siddseethepalli) [reply assistant/src/memory/v2/prompts/router.ts:80]: Addressed in https://github.com/vellum-ai/vellum-assistant/pull/30539

[the remark this answers is not in the document]
Comment (siddseethepalli) [reply assistant/src/memory/v2/prompts/router.ts:80]: Addressed in https://github.com/vellum-ai/vellum-assistant/pull/30539
